api/src/main/java/org/eclipse/microprofile/reactive/messaging/OnOverflow.java

```
@@ -0,0 +1,102 @@
+
+import static java.lang.annotation.ElementType.CONSTRUCTOR;
+import static java.lang.annotation.ElementType.FIELD;
+import static java.lang.annotation.ElementType.METHOD;
+import static java.lang.annotation.ElementType.PARAMETER;
+
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+
+
+/**
+ * Allows configuring the back pressure policy on injected {@link Emitter}:
+ *
+ * <pre>
+ * {
+ *     &#64;code
+ *     &#64;Channel("channel")
+ *     &#64;OnOverflow(value = OnOverflow.Strategy.BUFFER, bufferSize = 256)
+ *     Emitter&#60;String&#62; emitter;
+ * }
+ * </pre>
+ *
+ * <p>
+ * When not used, a {@link OnOverflow.Strategy#BUFFER} strategy is used with a buffer limited to 128 elements.
+ */
+@Retention(RetentionPolicy.RUNTIME)
+@Target({ METHOD, CONSTRUCTOR, FIELD, PARAMETER })
+public @interface OnOverflow {
+
+    /**
+     * The back pressure strategy.
+     */
+    enum Strategy {
+        /**
+         * Buffers <strong>all</strong> values until the downstream consumes it.
+         * This creates a buffer with the size specified by {@link #bufferSize()} if present.
+         * Otherwise, the size will be the value of the config property
+         * <strong>mp.messaging.emitter.default-buffer-size</strong>.
+         * If the buffer is full, an error will be propagated.
+         */
+        BUFFER,
+        /**
+         * Buffers <strong>all</strong> values until the downstream consumes it.
+         * This creates an unbound buffer. If the buffer is full, the application will die of {@code OutOfMemory}.
+         */
+        UNBOUNDED_BUFFER,
+        /**
+         * Drops the most recent value if the downstream can't keep up. It means that new value emitted by the upstream
+         * are ignored.
+         */
+        DROP,
+
+        /**
+         * Propagates a failure in case the downstream can't keep up.
+         */
+        FAIL,
```

Does this propagte an error to the subscriber by sending an `onError` signal, or does it cause an exception to be thrown back up to the caller from the `Emitter.onNext()` call?

Either would make sense, but it would be helpful to be clear which we expect to happen.

FWIW: the word "propagate" makes me think this is sending an error downstream. However, `Emitter.isRequested()` says that you can get an exception when you call `send` if the subscriber isn't ready to receive, but I don't see another backpressure strategy which would do that.

---

it should propagate an error to the subscriber by sending onError.